llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -9130,8 +9102,9 @@ void LoopVectorizationPlanner::adjustRecipesForReductions(
         VPInstruction::ComputeReductionResult, {PhiR, NewExitingVPV}, ExitDL);
     FinalReductionResult->insertBefore(*MiddleVPBB, IP);
     OrigExitingVPV->replaceUsesWithIf(
-        FinalReductionResult,
-        [](VPUser &User, unsigned) { return isa<VPLiveOut>(&User); });
+        FinalReductionResult, [](VPUser &User, unsigned) {
+          return isa<VPLiveOut>(&User) || isa<VPScalarStoreRecipe>(&User);
```

Comment:
better to also assert that VPScalarStoreRecipe is outside the vector loop region?